source/common/http/header_utility.cc

```diff
@@ -314,5 +314,11 @@ bool HeaderUtility::isRemovableHeader(absl::string_view header) {
          !absl::EqualsIgnoreCase(header, Headers::get().HostLegacy.get());
 }
 
+bool HeaderUtility::isModifiableHeader(absl::string_view header) {
+  return (header.empty() || header[0] != ':') &&
```

Comment:
Hm, there's a min value check in HeaderValueOption but as I'm not 100% sure validate() is called on every message including a HeaderValueOption (I've been bitten by validate() not being called so not being enforced before) I'd be inclined to leave it as an explicit check just in case.